presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -169,7 +170,9 @@ public void checkCanShowTables(ConnectorSecurityContext context, String schemaNa
     @Override
     public Set<SchemaTableName> filterTables(ConnectorSecurityContext context, Set<SchemaTableName> tableNames)
     {
-        return tableNames;
+        return tableNames.stream()
+                .filter(tableName -> checkAnyTablePermission(context, tableName))
```

Comment:
you need to also check if user has any access to the schema

Reply:
`checkAnyTablePermission` checks catalog access internally.  FWIU there is no schema `access` rule, and only a schema `owner` rule.

Reply:
I understand now.  I changed this to be `isSchemaOwner() || checkAnyTablePermission()`, so schema owners can see all the table names.